src/dswx_sar/masking_with_ancillary.py

```diff
@@ -349,51 +352,151 @@ def split_extended_water_parallel(
-    # bright water vs dark water
-    # water vs land
-    # Prepare arguments for parallel processing
+    temp_prefix = 'split_extended_water_parallel_temp'
+    temp_path_set = []
     minimum_pixel = 5000
-    args_list = [(i, sizes[i], minimum_pixel, coord_list[i],
-                  input_dict['intensity'],
-                  water_label_str,
-                  input_dict['water_mask'],
-                  pol_ind) for i in range(len(sizes))]
-
-    # check if the objects have heterogeneous characteristics (bimodality)
-    # If so, split the objects using multi-otsu thresholds and check bimodality.
-    results = Parallel(n_jobs=number_workers)(delayed(check_water_land_mixture)(args)
-                                  for args in args_list)
-
-    # If water need to be refined (change_flat=True), then update the water mask.
-    for i, change_flag, water_mask_subset in results:
-        if change_flag:
-            water_mask[coord_list[i][2] : coord_list[i][3],
-                       coord_list[i][0] : coord_list[i][1]] = water_mask_subset
 
-    dswx_sar_util.save_dswx_product(
-        np.array(water_mask, dtype=np.uint8),
-        output_path,
-        geotransform=meta_info['geotransform'],
-        projection=meta_info['projection'],
-        scratch_dir=outputdir
-        )
+    for block_iter, lines_per_block in enumerate(lines_per_block_set):
+        block_params = dswx_sar_util.block_param_generator(
+            lines_per_block,
+            [meta_info['length'], meta_info['width']],
+            pad_shape)
+        if len(lines_per_block_set) > 1:
+            removed_false_water_path = os.path.join(
+                outputdir,
+                f'{temp_prefix}_{block_iter}.tif')
+        else:
+            removed_false_water_path = output_path
+
+        temp_path_set.append(removed_false_water_path)
+
+        for block_ind, block_param in enumerate(block_params):
+            logger.info(
+                f'split_extended_water_parallel block #{block_ind} '
+                f'from {block_param.read_start_line} to '
+                f'{block_param.read_start_line + block_param.read_length}')
+            water_mask = dswx_sar_util.get_raster_block(
+                water_mask_path, block_param)
+            intensity_block = dswx_sar_util.get_raster_block(
+                input_dict['intensity'], block_param)
+
+            # Extract bounding boxes with buffer
+            coord_list, sizes, label_image = extract_bbox_with_buffer(
+                binary=water_mask, buffer=10)
+
+            filtered_sizes = []
+            filtered_coord_list = []
+            filtered_index = []
+            check_output = np.ones([len(sizes)], dtype='byte')
+            old_val = np.arange(1, len(sizes) + 1) - .1
+            index_array_to_image = np.array(
+                np.searchsorted(old_val, label_image),
+                dtype='uint32')
+
+            for ind in range(0, len(sizes)):
```

Comment:
nit - removed `0` since it is the default start.
```suggestion
            for ind in range(len(sizes)):
```